src/main/kotlin/org/cascadebot/cascadebot/commands/moderation/MuteChannelSetupSubCommand.kt

```diff
@@ -0,0 +1,121 @@
+import net.dv8tion.jda.api.entities.GuildChannel
+import net.dv8tion.jda.api.entities.Member
+import net.dv8tion.jda.api.entities.TextChannel
+import net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+import org.cascadebot.cascadebot.commandmeta.CommandContext
+import org.cascadebot.cascadebot.commandmeta.SubCommand
+import org.cascadebot.cascadebot.messaging.MessageType
+import org.cascadebot.cascadebot.messaging.Messaging
+import org.cascadebot.cascadebot.permissions.CascadePermission
+import org.cascadebot.cascadebot.utils.ConfirmUtils
+import org.cascadebot.cascadebot.utils.getMutedRole
+import java.util.concurrent.CompletableFuture
+
+class MuteChannelSetupSubCommand : SubCommand() {
+
+    override fun onCommand(sender: Member, context: CommandContext) {
+        if (!context.selfMember.hasPermission(Permission.MANAGE_CHANNEL)) {
+            context.uiMessaging.sendBotDiscordPermError(Permission.MANAGE_CHANNEL)
+            return
+        } else if (!context.selfMember.hasPermission(Permission.MANAGE_PERMISSIONS)) {
+            context.uiMessaging.sendBotDiscordPermError(Permission.MANAGE_PERMISSIONS)
+            return
+        }
+
+        if (ConfirmUtils.hasConfirmedAction("mute.channelsetup", context.user.idLong)) {
+            ConfirmUtils.completeAction("mute.channelsetup", context.user.idLong)
+            return
+        }
+
+        ConfirmUtils.confirmAction(
+                context.user.idLong,
+                "mute.channelsetup",
+                context.channel,
+                MessageType.WARNING,
+                context.i18n("commands.mute.channelsetup.warning"),
+                true,
+                object : ConfirmUtils.ConfirmRunnable() {
+                    override fun execute() {
+                        channelSetup(context)
+                    }
+                }
+        )
+    }
+
+    override fun command(): String = "channelsetup"
+
+    override fun parent(): String = "mute"
+
+    override fun permission(): CascadePermission = CascadePermission.of("mute.channelsetup", false)
+
+    private fun channelSetup(context: CommandContext) {
+        val mutedRole = context.guild.getMutedRole()
+        val success = mutableListOf<GuildChannel>()
+        val failed = mutableListOf<GuildChannel>()
+
+        val futures: MutableList<CompletableFuture<Void>> = mutableListOf()
+
+        context.guild.channels.forEach {
+            try {
+                futures.add(when (it) {
+                    is TextChannel -> it.manager.putPermissionOverride(mutedRole, 0, Permission.MESSAGE_WRITE.rawValue).submit()
+
+                    is Category -> {
+                        // Don't modify voice only categories
+                        if (it.textChannels.isEmpty()) return@forEach
+                        it.manager.putPermissionOverride(mutedRole, 0, Permission.MESSAGE_WRITE.rawValue).submit()
+                    }
+
+                    else -> return@forEach
```

Comment:
We should also remove speak permissions in vcs for the user and/or server mute them. I don't see ether of those being done right now.

Reply:
Hmm 🤔 Not sure how I feel about the `;mute` command affecting speaking.

Reply:
I personally would expect the mute command to also mute then in voice, and I'm sure other people would as well.

Reply:
Also remember you can't server mute people if they are not in voice, you'll recieve bad request. Same goes with unmuting.

Reply:
I expected that might be the case, thanks Ikinon! :)